pointercrate-user/src/auth/oauth2/get.rs

```diff
@@ -0,0 +1,71 @@
+use super::OAuth2AuthenticatedUser;
+use crate::auth::oauth2::GoogleUserInfo;
+use crate::config;
+use crate::error::UserError;
+use crate::{Result, User};
+use jsonwebtoken::{Algorithm, DecodingKey, Validation};
+use pointercrate_core::error::CoreError;
+use serde::Deserialize;
+use sqlx::{Error, PgConnection};
+
+#[derive(Deserialize)]
+struct GoogleTokenResponse {
+    id_token: String,
+}
+
+impl OAuth2AuthenticatedUser {
+    pub async fn by_google_account(id: &str, connection: &mut PgConnection) -> Result<Self> {
+        let row = sqlx::query!(
+            r#"SELECT member_id, members.name, CAST(permissions AS integer), display_name, youtube_channel::text, password_hash FROM members WHERE google_account_id = $1"#,
+            id
+        )
+        .fetch_one(connection)
+        .await;
+
+        match row {
+            Err(Error::RowNotFound) => Err(UserError::UserNotFoundGoogleAccount {
+                google_account_id: id.to_string(),
+            }),
+            Err(err) => Err(err.into()),
+            Ok(row) => Ok(OAuth2AuthenticatedUser {
+                user: construct_from_row!(row),
+                b64_salt: row.password_hash,
+                google_account_id: id.to_string(),
+            }),
+        }
+    }
+}
+
+#[cfg(feature = "oauth2")]
+pub async fn get_oauth2_id(code: &str) -> Result<GoogleUserInfo> {
+    let client = reqwest::Client::new();
+
+    let response = client
+        .post("https://oauth2.googleapis.com/token")
+        .form(&[
+            ("code", code),
+            ("client_id", &config::google_client_id()),
+            ("client_secret", &config::google_client_secret()),
+            ("redirect_uri", &config::google_redirect_uri()),
```

Comment:
I think this needs to be set to the same value as the `redirect_uri` field in the original request that returned the code in the first place.